matrix/rules.py

```diff
@@ -313,6 +313,15 @@ def exception_handler(self, context, loop=None, exc_ctx=None):
             return
         self._reported = True
 
+    async def model_change(self, delta, old_obj, new_obj, juju_model):
+        self.bus.dispatch(
```

Comment:
Well, it already has event support with the observer model.  This bit is really just translating from those observers to matrix's bus system.  Unifying those definitely seems like a good idea.